### ISSUE TITLE
Test mutation of ctor arguments

### ISSUE THREAD

**Issue by dmfs** [saynotobugsorg/confidence#73]
At present, it's cumbersome to test mutation of constructor arguments of mutable classes.

Consider the following class

```java
public final class ListAdder<T> implements Consumer<T> {
    private final List<T> list;

    public ListAdder(List<T> list) {
        this.list = list;
    }

    public void accept(T element) {
        list.add(element);
    }
}
```

**Comment by dmfs** [comment]
something like this might do the trick:

```java
assertThat(param->new ListAdder(param),
     mutatesParameter(
          ArrayList::new,
          soIt(iterates("a")),
          when(sucessfully(adder->adder.accept("a")))));
```

**Comment by dmfs** [comment]
closed by #95
